### PR TITLE
Feature/red 117 auto submit

### PR DESCRIPTION
**Pull request by red-tt-bear**

### CODE REVIEW

File: src/features/courses/course-controller.ts [hunk 1]
```diff
@@ -2717,62 +2722,90 @@ class CourseController {
...
+            const startTime = moment().add(1,'minute'); // 1 minute should cover any delay in creating records before a student can actually begin
+    
+            let endTime = moment(startTime).add(topicInfo.duration, 'minutes');
+            if (topicInfo.hardCutoff) {
+                endTime = moment.min(topic.endDate.toMoment(), endTime);
+            } 
+    
+            const nextVersionAvailableTime = moment(startTime).add(topicInfo.versionDelay, 'minutes');
+            // in trying to be clever about shuffling the order, don't forget to shift from 0..n-1 to 1..n
+            const problemOrder = (topicInfo.randomizeOrder) ? _.shuffle([...Array(10).keys()]) : [...Array(10).keys()];
```
Comment (PanopticaRising) [inline src/features/courses/course-controller.ts:2740]: Why is this hardcoded to 10?

File: src/routes.ts [hunk 2]
```diff
@@ -44,7 +44,7 @@ router.use('/version', expressAsyncHandler(async (_req: express.Request, _res: e
         packageJson: version
     }));
 }));
-// TODO remove
-router.use('/schedule', require('./scheduler-routes'));
+// Dev routes for testing the scheduler, if your using this remember to uncomment the white lister for it as well in server.ts limiter
```
Comment (doombot-exe) [inline src/routes.ts:47]: you're for 'you are' -- not 'your'

**Review by PanopticaRising — APPROVED** [review]

File: src/features/courses/course-controller.ts (hunk 1 above)
Comment (red-tt-bear) [reply src/features/courses/course-controller.ts:2740]: This isn't actually one of my changes, this is already in dev, if you check without whitespace this was just the transaction fix

Comment (PanopticaRising) [reply src/features/courses/course-controller.ts:2740]: This might be @doombot-exe 's code

File: src/routes.ts (hunk 2 above)
Comment (red-tt-bear) [reply src/routes.ts:47]: 🤣 good catch

**Review by doombot-exe — APPROVED** [review]

**Pull request merged**